### PR TITLE
Separate port for readiness check prior to warmup

### PR DESCRIPTION
**Pull request by worldtiki**
Just a draft at this point.

We recently ran into an issue where we had our management endpoints (/health, /info, etc) served by a separate port. This change would allow users of mittens to, for example, wait for the /health to be ready in port 8081 and then run warmup requests against port 8080.

### CODE REVIEW

File: pkg/warmup/target.go [hunk 1]
```diff
@@ -57,15 +60,16 @@ func NewTarget(httpClient whttp.Client, grpcClient grpc.Client, options TargetOp
...
+	log.Printf("waiting for %d:%s target readiness probe", t.options.ReadinessPort, t.options.ReadinessPath)
 	for {
 		select {
 		case <-time.After(time.Duration(t.options.ReadinessTimeoutInSeconds) * time.Second):
 			return fmt.Errorf("target readiness proble: timeout %d seconds exceeded", t.options.ReadinessTimeoutInSeconds)
 		case <-done:
 			return errors.New("target readiness probe: received done signal")
 		default:
-			if err := t.httpClient.Request(http.MethodGet, t.options.ReadinessPath, nil, nil); err != nil {
+			// FIXME: just http here? no grpc?
```
Comment (worldtiki) [inline pkg/warmup/target.go]: this bit here seems odd. this file is not http specific but there's only one branch for http (none for grpc)

File: docs/about/getting-started.md [hunk 2]
```diff
@@ -13,31 +13,32 @@ The application receives a number of command-line flags including the requests t
...
+| --target-http-port                 | int    | 8080                     | Http port for warm up requests                                                                                                                                                    |
+| --target-insecure                  | bool   | false                    | Whether to skip TLS validation                                                                                                                                                    |
+| --target-readiness-path            | string | /ready                   | The path used for target readiness probe                                                                                                                                          |
+| --target-readiness-port            | int    | 8080                     | The port used for the readiness probes                                                                                                                                            |
```
Comment (worldtiki) [inline docs/about/getting-started.md]: added this one

File: pkg/warmup/target.go (hunk 1 above)
Comment (nikos912000) [reply pkg/warmup/target.go]: Mittens doesn't support gRPC health checks atm. Would you mind raising an issue and we can look into it?

**Comment by nikos912000** [comment]
Nice, thanks for this!

lgtm. As discussed internally, I'm raising a PR to drop cobra and use flags only. Then we can merge this one.

File: go.mod [hunk 3]
```diff
@@ -1,9 +1,9 @@
 module mittens
 
 require (
```
Comment (nikos912000) [inline go.mod]: Can you please run `go mod tidy` to clean up the `go.mod` and `go.sum` files?
You need to have Go modules enabled which you can do with `GO111MODULE=on`.

Comment (nikos912000) [reply go.mod]: Thanks!

**Pull request merged**